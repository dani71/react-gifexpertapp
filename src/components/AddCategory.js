import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    //const [inputValue, setInputValue] = useState();// esto da error, porque es undefined
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        // console.log(e.target.value);
        setInputValue(e.target.value);
        // console.log('HandelInputChange llamado');
    }

    const handleSubmit = (e) => {  
        e.preventDefault();
        if (inputValue.trim().length > 2) { //evita la categoría vacía
            // console.log(setCategories)
            //setCategories((c) => {[...c, e]});
            setCategories((cats) => [inputValue, ...cats]);
            setInputValue(''); // evita el escribir dos veces lo mismo initencionadamente
        };

    }
    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}