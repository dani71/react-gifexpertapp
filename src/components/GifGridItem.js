import React from 'react'

export const GifGridItem = ( {id, title, url} ) => {
 
    console.log({id, title, url})
    return (
        <div className="card animate__animated animate__fadeInDown"> {/*class puede ser para html o para JS, as√≠ que la han cambiado a className para estar seguros*/}
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
// class="animate__animated animate__bounce animate__delay-2s"
