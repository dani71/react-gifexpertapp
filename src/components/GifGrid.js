import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import React, { useState, useEffect } from 'react'
// import { getGifs } from '../helpers/getGifs'
import { GifGridItem } from './GifGridItem'


export const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([])

    // useEffect(() => {
    //     getGifs( category )
    //     .then( setImages ); // esto es lo mismo que .then( imgs => setImages( imgs ))
    // }, [ category ]) //aquí se pasa una lista de dependencias, y si no tiene no lanza otras peticiones 
    //                     // si la categoría cambia se dispara otra vez.    

    const { data: images, loading } = useFetchGifs(category);

    // console.log(loading);
    // console.log(data);

    return (
        <>
            <h3 className="animate__animated animate__fadeInDown">{category}</h3>
            {loading && <p className='animate__animated animate__flash'>Loading</p> /*solo hace algo si está cargando*/}
            <div className='card-grid'>
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
