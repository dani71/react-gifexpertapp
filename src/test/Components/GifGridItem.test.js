import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import { GifGridItem } from '../../components/GifGridItem';

// import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Un título';
    const url = 'http://localhost/algo.jpg'

    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('debe de mostrar el componente correctamente', () => {
        // const wrapper = shallow(<GifGridItem title={title} url={url} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un párrafo con el título', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('debe tene la imagen igual al url y al de los propos ', () => {

        const img = wrapper.find('img');
        // console.log(img.html())
        // console.log(img.props())
        // console.log(img.prop('src'))
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe tener la clase animate__fadeIn', () => {

        const div = wrapper.find('div')
        const className = div.prop('className');

        expect(div.prop('className')).toContain('animate__fadeInDown');
        expect(className.includes('animate__fadeInDown')).toBe(true);       //es lo mismo pero de otra manera
    });

});
