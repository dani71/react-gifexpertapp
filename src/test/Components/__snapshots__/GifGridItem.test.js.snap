// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
   
  <img
    alt="Un título"
    src="http://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
